test/mux.sh

```diff
@@ -408,35 +418,64 @@ if [ "$MUX_FILE_BUFFER_STORAGE_TYPE" = "pvc" -o "$MUX_FILE_BUFFER_STORAGE_TYPE"
 
     update_current_fluentd $ENABLE_SECURE_FORWARD
 
-    ES_HOST_BAK=$( oc set env --list dc/logging-mux | grep \^ES_HOST= )
-    OPS_HOST_BAK=$( oc set env --list dc/logging-mux | grep \^OPS_HOST= )
-
-    # set ES_HOST and OPS_HOST to bogus
-    reset_ES_HOST ES_HOST=bogus OPS_HOST=bogus
+    # set ES_HOST and OPS_HOST to non-existing hostname
+    if [ "$ops_cluster" = "true" ]; then
+        reset_ES_HOST ES_HOST=bogus OPS_HOST=bogus_ops
+    else
+        reset_ES_HOST ES_HOST=bogus OPS_HOST=bogus
+    fi
 
     uuid_es=$( uuidgen )
     uuid_es_ops=$( uuidgen )
 
     logger -i -p local6.info -t $uuid_es_ops $uuid_es_ops
     add_test_message $uuid_es
 
-    # wait long enough to make the test messages are in the buffer
-    sleep 10
-    muxpod=$( oc get pods -l component=mux -o name | awk -F'/' '{print $2}' )
+    # wait for the test messages are in the buffer
+    muxpod=$( get_running_pod mux )
+    if [ "$ops_cluster" = "true" ]; then
+       ops_filename="output-es-ops-config.output_ops_tag"
+    else
+       ops_filename="output-es-config.output_tag"
+    fi
+    retry="true"
+    while [ "$retry" = "true" ]; do
+        ops_logs=$( oc exec $muxpod -- ls /var/lib/fluentd/ | egrep $ops_filename ) || :
+        if [ -z "$ops_logs" ]; then
+            sleep 1
+            continue
+        fi
+        for ops_log in $ops_logs; do
+            found=$( oc exec $muxpod -- strings /var/lib/fluentd/$ops_log | egrep $uuid_es_ops ) || :
+            if [ -n "$found" ]; then
+                retry="false"
+                break
+            fi
+        done
+    done
     os::log::debug "$( oc exec $muxpod -- ls -l /var/lib/fluentd )"
     os::log::debug "$( oc logs $muxpod )"
 
     # set ES_HOST and OPS_HOST to original
     reset_ES_HOST $ES_HOST_BAK $OPS_HOST_BAK
 
+    # wait for the file buffer disappears once
+    os::cmd::try_until_text "oc exec $muxpod -- ls -l /var/lib/fluentd" "total 0" $FLUENTD_WAIT_TIME
+    os::log::debug "$( oc exec $muxpod -- ls -l /var/lib/fluentd )"
+    os::log::debug "$( oc logs $muxpod )"
+
     # kibana logs with kibana container/pod values
     myproject=project.logging
-    mymessage=$uuid_es
-    os::cmd::try_until_success "curl_es $es_pod /${myproject}.*/_count?q=message:$mymessage | get_count_from_json | grep -q 1" "$(( 10*minute ))"
+    mymessage="GET /${uuid_es} 404 "
+    qs='{"query":{"match_phrase":{"message":"'"${mymessage}"'"}}}'
+    os::log::debug "Check kibana log - message \"${mymessage}\""
+    os::cmd::try_until_success "curl_es $es_pod /${myproject}.*/_count -XPOST -d '$qs' | get_count_from_json | grep -q 1" "$(( 10*minute ))"
```

Comment:
do you want to test if the output from count contains `1` (e.g. `100`, `21` all match `grep -q 1`) or do you want to match exactly `1` e.g. `grep -q '^1$'`?

Reply:
Good catch!  Pushed the fix.  (Adding the fix to pr#756 for master, too)